src/Bicep.Core/Analyzers/Linter/Rules/NoHardcodedEnvironmentUrlsRule.cs

```diff
@@ -36,56 +44,56 @@ public override void Configure(IConfigurationRoot config)
         public override string FormatMessage(params object[] values)
             => string.Format("{0} Found this disallowed host: \"{1}\"", this.Description, values.First());
 
+        private Regex CreateDisallowedHostRegex() =>
+            new Regex(string.Join('|', this.DisallowedHosts), RegexOptions.Compiled | RegexOptions.IgnoreCase);
+
+        private Regex CreateSchemaRegex() =>
+            new Regex(@"https://schema\.|http://schema\.", RegexOptions.Compiled | RegexOptions.RightToLeft | RegexOptions.IgnoreCase);
```

Comment:
I don't think it makes sense to keep this unconfigurable logic from the TTK. The only excluded host that I'm aware of is `schema.management.azure.com`, and I don't think this rule translates generically to any host beginning with `schema.`. The other change to have a specific list of disallowed hosts covers this scenario adequately.

Reply:
this has been replaced with exclusion list